File: src/Models/Aluno.java
package Models;

import java.time.LocalDateTime;

public class Aluno extends Usuario{
    public Aluno(String nome, String email, String id) {
        super(nome, email, id);
    }

    @Override
    public boolean podeReservarComAntecedencia(LocalDateTime data) {
        return data.isBefore(LocalDateTime.now().plusHours(24));
    }
}

File: refactor-python/Models/Aluno.py
import datetime

from Models.Usuario import Usuario


class Aluno(Usuario):
    def __init__(self, nome, id, email):
        super().__init__(id, nome, email)

    def podeReservarComAntecedencia(self, dataHoraReserva):
        dataLimite = datetime.now() + datetime.timedelta(hours=24)
        return dataHoraReserva < dataLimite